src/cephadm/cephadmlib/daemons/ceph.py

```diff
@@ -407,6 +410,55 @@ def create_daemon_dirs(self, data_dir: str, uid: int, gid: int) -> None:
         populate_files(ssl_dir, cert_files, uid, gid)
 
 
+def retrieve_version(container_id: str, container_command: str) -> str:
+    try:
+        exec_command = [
+            container_command,
+            'exec',
+            container_id,
+            'python3',
+            '-c',
+            "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')",
+        ]
+
+        exec_result = subprocess.run(
+            exec_command, capture_output=True, text=True, check=True
+        )
+        return exec_result.stdout.strip()
+
+    except Exception as e:
+        logger.error(
+            f'Failed to retrieve the python version from container, failed with: {str(e)}'
+        )
+    return ''
+
+
+def get_python_version(ctx: CephadmContext) -> str:
+    try:
+        from cephadm import get_container_info
```

Comment:
This is a little odd. I know you're trying to work around a dependency issue but I'm not sure if this is the best way to go about it.  However, I'm not going to outright block this PR over if if we can make this a temporary thing... let's think on it some more.

Reply:
FWIW, this patch as well as worked I started for https://tracker.ceph.com/issues/69278 is what motivated this refactor. an initial PR should be up very soon.

Reply:
@Kushal-deb FYI the patches we were talking about in this thread have been merged, so you could take another look at this when you have bandwidth